onnxruntime/core/mlas/lib/amd64/QLinearBinaryOpKernelAvx2.asm

```diff
@@ -0,0 +1,436 @@
+        add     rcx,rax                         ; out-of-order instruction(s)
+        sub     rdx,rax                         ; out-of-order instruction(s)
+        vcvtps2dq ymm8,ymm8                     ; nearbyintf()()
+        vpmaxsd ymm8,ymm8,ymm7
+        vpminsd ymm8,ymm8,ymm6
+        vpaddd  ymm8,ymm8,ymm5                  ; + ZeroPointC
+        vpshufb ymm8,ymm8,ymm10                 ; pack 32bits integers into 8bit integers
+        vpermps ymm8,ymm11,ymm8
+
+        jb      QLinear&OpName&&DataType&Avx2StoreLessThan8VectorOnScalar
+
+        movsd   QWORD PTR [r8],xmm8
+        add     r8,rax
+        test    rdx,rdx
+        jz      QLinear&OpName&&DataType&Avx2Exit
+        jmp     QLinear&OpName&&DataType&Avx2Process8LoopVectorOnScalar
+
+QLinear&OpName&&DataType&Avx2StoreLessThan8VectorOnScalar:
+        add     rdx,rax
+        pextrq  rax,xmm8,0
+        jmp     QLinear&OpName&&DataType&Avx2StoreLoopGeneral
+
+
+QLinear&OpName&&DataType&Avx2Process8LoopVectorOnVector:
+        test    rdx,rdx
+        jz      QLinear&OpName&&DataType&Avx2Exit
+
+        UnpackBytesDWords ymm8,[rcx],DataType   ; IntegerVectorA
+        UnpackBytesDWords ymm9,[r9],DataType    ; IntegerVectorB
+        vpsubd  ymm8,ymm8,ymm3                  ; - ZeroPointA
+        vpsubd  ymm9,ymm9,ymm4                  ; - ZeroPointB
+        vcvtdq2ps ymm8,ymm8                     ; FloatVectorA
+        vcvtdq2ps ymm9,ymm9                     ; FloatVectorB
+        vmulps  ymm8,ymm8,ymm0                  ; * ScaleA
+        vmulps  ymm9,ymm9,ymm1                  ; * ScaleB
+
+        OpInstruction  ymm8,ymm8,ymm9           ; OpName two float values
+
+        vdivps  ymm8,ymm8,ymm2                  ; Quantize 8 values, / ScaleC
+        add     rcx,rax                         ; out-of-order instruction(s)
+        add     r9,rax                          ; out-of-order instruction(s)
+        sub     rdx,rax                         ; out-of-order instruction(s), set flag for jb below
+        vcvtps2dq ymm8,ymm8                     ; nearbyintf()
+        vpmaxsd ymm8,ymm8,ymm7
+        vpminsd ymm8,ymm8,ymm6
+        vpaddd  ymm8,ymm8,ymm5                  ; + ZeroPointC
+        vpshufb ymm8,ymm8,ymm10                 ; pack 32bits integers into 8bit integers
+        vpermps ymm8,ymm11,ymm8
+
+        jb      QLinear&OpName&&DataType&Avx2StoreLessThan8VectorOnVector
+
+        movsd   QWORD PTR [r8],xmm8
+        add     r8,rax
+        jmp     QLinear&OpName&&DataType&Avx2Process8LoopVectorOnVector
+
+QLinear&OpName&&DataType&Avx2StoreLessThan8VectorOnVector:
+        add     rdx,rax
+        pextrq  rax,xmm8,0
+
+QLinear&OpName&&DataType&Avx2StoreLoopGeneral:
+        mov     BYTE PTR [r8],al
+        shr     rax,8
+        inc     r8
+        dec     rdx
+        jnz     QLinear&OpName&&DataType&Avx2StoreLoopGeneral
+
+QLinear&OpName&&DataType&Avx2Exit:
+        vzeroupper
+        vmovaps xmm6,QLinearBinaryOpFrame.SavedXmm6[rsp]
```

Comment:
for the xmm restore, as with save_xmm above, I was switching these back to plain movaps

Reply:
Sure! Fixed!